<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>824</width>
    <height>537</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SLStudio</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="acceptDrops">
    <bool>false</bool>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="PointCloudWidget" name="pointCloudWidget" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>824</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuCalibration">
    <property name="title">
     <string>Calibration</string>
    </property>
    <addaction name="actionLoadCalibration"/>
    <addaction name="actionExportCalibration"/>
    <addaction name="actionCalibration"/>
   </widget>
   <widget class="QMenu" name="menuSLStudio">
    <property name="title">
     <string>SLStudio</string>
    </property>
    <addaction name="actionPreferences"/>
    <addaction name="actionQuit"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuScan">
    <property name="title">
     <string>Scan</string>
    </property>
    <addaction name="actionStart"/>
    <addaction name="actionStop"/>
    <addaction name="actionSavePointCloud"/>
    <addaction name="actionSaveScreenshot"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionUpdatePointClouds"/>
   </widget>
   <widget class="QMenu" name="menuProjector">
    <property name="title">
     <string>Projector</string>
    </property>
    <addaction name="actionUpload_Scan_Patterns"/>
    <addaction name="actionUpload_Calibration_Patterns"/>
   </widget>
   <addaction name="menuSLStudio"/>
   <addaction name="menuScan"/>
   <addaction name="menuCalibration"/>
   <addaction name="menuProjector"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="font">
    <font>
     <italic>false</italic>
     <strikeout>false</strikeout>
    </font>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionStart"/>
   <addaction name="actionStop"/>
   <addaction name="separator"/>
   <addaction name="actionSavePointCloud"/>
   <addaction name="actionSaveScreenshot"/>
   <addaction name="actionCalibration"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionQuit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="actionStart">
   <property name="text">
    <string>Start Scan</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Stop Scan</string>
   </property>
  </action>
  <action name="actionCalibration">
   <property name="text">
    <string>Perform Calibration</string>
   </property>
  </action>
  <action name="actionLoadCalibration">
   <property name="text">
    <string>Load Calibration</string>
   </property>
  </action>
  <action name="actionExportCalibration">
   <property name="text">
    <string>Export Calibration</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="text">
    <string>&amp;Preferences</string>
   </property>
  </action>
  <action name="actionSavePointCloud">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save Point Cloud</string>
   </property>
  </action>
  <action name="actionSaveScreenshot">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save Screenshot</string>
   </property>
  </action>
  <action name="actionTracking">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Start Tracking</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
  <action name="actionTracker">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Tracker</string>
   </property>
  </action>
  <action name="actionUpdatePointClouds">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Update Point Clouds</string>
   </property>
  </action>
  <action name="actionUpload_Calibration_Patterns">
   <property name="text">
    <string>Upload Calibration Patterns</string>
   </property>
  </action>
  <action name="actionUpload_Scan_Patterns">
   <property name="text">
    <string>Upload Scan Patterns</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PointCloudWidget</class>
   <extends>QWidget</extends>
   <header>PointCloudWidget.h</header>
   <container>1</container>
   <slots>
    <slot>saveScreenShot()</slot>
    <slot>savePointCloud()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionStart</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionStart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStop</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionStop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>457</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCalibration</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionCalibration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreferences</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionPreferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExportCalibration</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionExportCalibration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveScreenshot</sender>
   <signal>triggered()</signal>
   <receiver>pointCloudWidget</receiver>
   <slot>saveScreenShot()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSavePointCloud</sender>
   <signal>triggered()</signal>
   <receiver>pointCloudWidget</receiver>
   <slot>savePointCloud()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoadCalibration</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionLoadCalibration()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onActionAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>268</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onActionStart()</slot>
  <slot>onActionStop()</slot>
  <slot>onActionCalibration()</slot>
  <slot>onActionPreferences()</slot>
  <slot>onActionExportCalibration()</slot>
  <slot>onActionTracker()</slot>
  <slot>onActionLoadCalibration()</slot>
  <slot>onActionHistogram()</slot>
  <slot>onActionShading()</slot>
  <slot>onActionAbout()</slot>
  <slot>onActionDecoderUp()</slot>
  <slot>onActionDecoderVp()</slot>
 </slots>
</ui>
